chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/impl/UserCertificationBodyMapDAOImpl.java

```diff
@@ -0,0 +1,162 @@
+package gov.healthit.chpl.dao.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.persistence.Query;
+
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.context.support.DefaultMessageSourceResolvable;
+import org.springframework.stereotype.Repository;
+
+import gov.healthit.chpl.auth.entity.UserEntity;
+import gov.healthit.chpl.auth.user.User;
+import gov.healthit.chpl.dao.UserCertificationBodyMapDAO;
+import gov.healthit.chpl.dto.UserCertificationBodyMapDTO;
+import gov.healthit.chpl.entity.CertificationBodyEntity;
+import gov.healthit.chpl.entity.UserCertificationBodyMapEntity;
+import gov.healthit.chpl.exception.EntityRetrievalException;
+
+@Repository(value = "userCertificationBodyMapDAO")
+public class UserCertificationBodyMapDAOImpl extends BaseDAOImpl implements UserCertificationBodyMapDAO {
+
+    @Override
+    public UserCertificationBodyMapDTO create(UserCertificationBodyMapDTO dto) throws EntityRetrievalException {
+        UserCertificationBodyMapEntity entity = new UserCertificationBodyMapEntity();
+        entity = create(getUserCertificationBodyMapEntity(dto));
+        return new UserCertificationBodyMapDTO(entity);
+    }
+
+    @Override
+    public UserCertificationBodyMapDTO update(UserCertificationBodyMapDTO dto) throws EntityRetrievalException {
+        UserCertificationBodyMapEntity entity = new UserCertificationBodyMapEntity();
+        entity = update(getUserCertificationBodyMapEntity(dto));
+        return new UserCertificationBodyMapDTO(entity);
+    }
+
+    public void delete(UserCertificationBodyMapDTO dto) throws EntityRetrievalException {
+        update(getUserCertificationBodyMapEntity(dto));
+    }
+
+    @Override
+    public List<UserCertificationBodyMapDTO> getByUserId(Long userId) {
+        Query query = entityManager.createQuery(
+                "from UserCertificationBodyMapEntity ucbm " + "join fetch ucbm.certificationBody cb "
```

Comment:
Since we are querying user_certification_body_map (not the user table) and that table requires (not null) a user_id and certification_body_id, a left join is not required.  I'm going to add a null check before populating the list of DTOs just to be safe, though.  This method will then be sure to always a return a `List<UserCertificationBodyMapDTO>` and never null.

Reply:
Since we a re querying the user_certification_body_map table which requires a user_id and certification_body_id, The query should simply return 0 results.  I'm adding a null check to ensure that we always return `List<UserCertificationBodyMapDTO>` and not null.